src/config.rs

```diff
@@ -0,0 +1,399 @@
+use anyhow::{bail, Result};
+
+/// Prints a debug message to the GitHub Actions log if `RUNNER_DEBUG` or `ACTIONS_RUNNER_DEBUG` are set.
+#[macro_export]
+macro_rules! debug {
+    ($($arg:tt)*) => {
+        if std::env::var("RUNNER_DEBUG").is_ok() || std::env::var("ACTIONS_RUNNER_DEBUG").is_ok() {
+            println!("::debug::{}", format!($($arg)*));
+        }
+    };
+}
+
+const EU_DEFAULT_API_URL: &str = "https://api.bitwarden.eu";
+const EU_DEFAULT_IDENTITY_URL: &str = "https://identity.bitwarden.eu";
+
+const US_DEFAULT_API_URL: &str = "https://api.bitwarden.com";
+const US_DEFAULT_IDENTITY_URL: &str = "https://identity.bitwarden.com";
+
+#[derive(Debug)]
+/// Input parameters for the GitHub Action.
+pub struct Config {
+    pub access_token: String,
+    pub secrets: Vec<String>,
+    pub cloud_region: String, // "US" or "EU"; default is "US"
+    pub base_url: Option<String>,
+    pub api_url: Option<String>,
+    pub identity_url: Option<String>,
+    pub set_env: bool,
+}
+
+impl Config {
+    /// Creates a new Config instance from environment variables.
+    pub fn new() -> Result<Self> {
+        let cloud_region = get_env("INPUT_CLOUD_REGION")
+            .unwrap_or_default()
+            .to_lowercase()
+            .trim()
+            .to_owned();
+
+        if cloud_region != "us" && cloud_region != "eu" && !cloud_region.is_empty() {
+            bail!("Cloud region must be either 'US' or 'EU'");
+        }
+
+        let access_token = get_env("INPUT_ACCESS_TOKEN")
+            .ok_or_else(|| anyhow::anyhow!("Access token is required"))?;
+
+        let secrets = get_env("INPUT_SECRETS")
+            .ok_or_else(|| anyhow::anyhow!("Secrets are required"))?
+            .lines()
+            .map(str::trim)
+            .filter(|s| !s.is_empty())
+            .map(String::from)
+            .collect();
+
+        let base_url = get_env("INPUT_BASE_URL");
+        let api_url = get_env("INPUT_API_URL");
+        let identity_url = get_env("INPUT_IDENTITY_URL");
+
+        validate_urls(
+            base_url.as_deref(),
+            api_url.as_deref(),
+            identity_url.as_deref(),
+        )?;
+
+        let set_env = get_env("INPUT_SET_ENV").is_some_and(|val| val != "false");
```

Comment:
nit: Could do ignore-case here. It's actually even what the typescript `getBooleanInput` would have done if we had this feature in that.